samples/Mvc.TagHelperSample.Web/Services/SignalTokenProviderService.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Threading;
+using Microsoft.Extensions.Primitives;
+
+namespace TagHelperSample.Web.Services
+{
+    public class SignalTokenProviderService : ISignalTokenProviderService
+    {
+        private readonly ConcurrentDictionary<object, ChangeTokenInfo> _changeTokens
+            = new ConcurrentDictionary<object, ChangeTokenInfo>();
+
+        public IChangeToken GetToken(object key)
```

Comment:
We only ever pass in string keys, might be worth simplifying this to express that.